### PR TITLE
Enhance dbinfo sub-command with column names and global variables

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
* Rename `schema` to `dbinfo`
* Add column names to returned info
* Add global variables (small subset for now)
## todos

- [ ] Add foreign key info (will be done in future PR)
- [x] Add test for getting data from a test cluster: currently testing against local mysql
- [x] Local linter is still failing, find fix and validate code style

### CODE REVIEW

File: go/dbinfo/dbinfo.go [hunk 1]
```diff
@@ -47,9 +45,18 @@ func run(out io.Writer, cfg Config) error {
 	return err
 }
 
+type TableColumn struct {
+	Name       string `json:"name"`
+	Type       string `json:"type"`
+	KeyType    string `json:"keyType"`
```
Comment (systay) [inline go/dbinfo/dbinfo.go:51]:
```suggestion
	KeyType    string `json:"keyType,omitEmpty"`
```

File: go/dbinfo/dbinfo.go [hunk 2]
```diff
@@ -47,14 +45,24 @@ func run(out io.Writer, cfg Config) error {
 	return err
 }
 
+type TableColumn struct {
+	Name       string `json:"name"`
+	Type       string `json:"type"`
+	KeyType    string `json:"keyType,omitempty"`
+	IsNullable bool   `json:"isNullable"`
```
Comment (systay) [inline go/dbinfo/dbinfo.go:52]: nitpick: could we slim up the json output? like removing `isNullable` for false values?

File: go/dbinfo/dbinfo.go [hunk 3]
```diff
@@ -47,14 +45,24 @@ func run(out io.Writer, cfg Config) error {
...
+	Rows    int            `json:"rows"`
+	Columns []*TableColumn `json:"columns"`
 }
 
 type Info struct {
-	FileType string      `json:"fileType"`
-	Tables   []TableInfo `json:"tables"`
+	FileType        string             `json:"fileType"`
+	Tables          []*TableInfo       `json:"tables"`
+	GlobalVariables *map[string]string `json:"globalVariables"`
```
Comment (systay) [inline go/dbinfo/dbinfo.go:65]: why a reference to a map? maps are already sent around by reference, so I'm confused to why

File: go/dbinfo/utils.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+			Extra:      extra,
+		}
+		tc[tableName] = append(tc[tableName], col)
+	}
+	return tc, nil
+}
+
+func (dbh *DBHelper) getGlobalVariables() (*map[string]string, error) {
+	// Currently only use simple regex to match the variable names
+	// If the variable name contains ".*" then it is treated as a regex, else exact match
```
Comment (systay) [inline go/dbinfo/utils.go:104]: seems premature to add the regex functionality, since we are not using it, or have any plans on using it at the moment

**Comment by systay** [comment]
we should also add something to the main README about this functionality

File: go/cmd/dbinfo.go [hunk 5]
```diff
@@ -20,31 +20,31 @@ import (
...
+			return dbinfo.Run(cfg)
 		},
 	}
 
 	cmd.Flags().StringVarP(&vtParams.Host, "host", "", "127.0.0.1", "Database host")
 	cmd.Flags().IntVarP(&vtParams.Port, "port", "", 3306, "Database port")
 	cmd.Flags().StringVarP(&vtParams.Uname, "user", "", "root", "Database user")
 	cmd.Flags().StringVarP(&vtParams.Pass, "password", "", "", "Database password")
-	cmd.Flags().StringVarP(&vtParams.DbName, "database", "", "", "Database name")
+	cmd.Flags().StringVarP(&vtParams.DbName, "database", "", "sakila", "Database name")
```
[the remark this answers is not in the document]
Comment (rohit-nayak-ps) [reply go/cmd/dbinfo.go]: Removed

File: go/dbinfo/utils.go (hunk 4 above)
Comment (rohit-nayak-ps) [reply go/dbinfo/utils.go:103]: removed

**Comment by rohit-nayak-ps** [comment]
> we should also add something to the main README about this functionality

Added some information there. PTAL and modify if required. Also I have auto-wrap setup for md files, if that is not desired I can revert and only update with my changes ...

**Review by systay — APPROVED** [review]

**Pull request merged**